### PR TITLE
Code Refactoring

### PR DESCRIPTION
**Pull request by miraclestyle**
Ticket:
- [Refactor entire frontend to follow all the styles and optimizations.](https://trello.com/c/vgIbYaML/45-refactor-entire-frontend-to-follow-all-the-styles-and-optimizations)

This branch bears a large change.

Animation and positioning logic of the frontend has not been altered from the previous code version. That remains as a separate ticket to be done if time permits.

### CODE REVIEW

File: client/app/components/lib/arrow-button.jsx [hunk 1]
```diff
@@ -63,28 +66,41 @@ const Button = style.button`
...
+    </Button>
+  );
+};
+
+ArrowButton.propTypes = {
+  title: PropTypes.string.isRequired,
+  side: PropTypes.string.isRequired,
+  hide: PropTypes.bool.isRequired,
+  click: PropTypes.func.isRequired,
+};
```
Comment (Matt-Wigg) [inline client/app/components/lib/arrow-button.jsx:104]: Nice!

File: client/app/components/product-card/index.jsx [hunk 2]
```diff
@@ -51,26 +42,69 @@ const ProductLink = style(Link)`
...
+    review_count: PropTypes.number.isRequired,
+  }),
+  quickview: PropTypes.bool.isRequired,
+  open: PropTypes.func.isRequired,
+  close: PropTypes.func.isRequired,
+};
+
+Card.defaultProps = {
+  product: PropTypes.object,
+};
```
Comment (Matt-Wigg) [inline client/app/components/product-card/index.jsx:108]: I think defaultProps should have an actual value - `( { hello: 'word' } )` for example  - not a propType.

File: client/app/components/product-card/product-name.jsx [hunk 3]
```diff
@@ -1,20 +1,30 @@
 const React = require('react');
+const PropTypes = require('prop-types');
 const style = require('styled-components').default;
-const Container = require('../lib/container.jsx');
-const Heading = require('../lib/heading.jsx');
 
-const NameContainer = style(Container)`
+const Container = style.div`
```
Comment (Matt-Wigg) [inline client/app/components/product-card/product-name.jsx:5]: NameContainer seemed like a better descriptive variable here!

File: client/app/components/product-card/index.jsx (hunk 2 above)
Comment (miraclestyle) [reply client/app/components/product-card/index.jsx:108]: I will look into that. Thanks for the advice.

File: client/app/components/product-card/product-name.jsx (hunk 3 above)
Comment (miraclestyle) [reply client/app/components/product-card/product-name.jsx:5]: I thought that too. I am still contemplating what to do about that. This short version is less verbose.

File: client/app/components/product-card/index.jsx (hunk 2 above)
Comment (miraclestyle) [reply client/app/components/product-card/index.jsx]: I fixed this. I had to require composites in type checks. If fine tunning necessitates, I will gradually provide defaults.

**Pull request merged**